examples/agent-sandbox-rl/agent_sandbox_rl/recycle.py

```
@@ -0,0 +1,476 @@
+              if not recyclable:
+                logger.info("image %s not git-recyclable — fresh claim per task", t.image)
+            elif recyclable:
+              baseline = await afleet._to_thread(reset.prime, handle)
+            if scale_on_hold and recyclable:     # holding → stop the pool replenishing
+              await afleet._to_thread(sync.unwarm_image, t.image)
+            uses = 0
+          await _process(handle, i, t)
+          if pos == len(group) - 1:
+            break
+          if not recyclable:                      # can't reset → fresh claim per task
+            await afleet.release(handle); handle = None; continue
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)